### ISSUE TITLE
Class-based ComponentMetatdataRule Ordering

### ISSUE THREAD

**Issue by sghill** [nebula-plugins/gradle-nebula-integration#23]
Switching over to class-based component metadata rules introduces this exception if everything isn't class-based:

```
Caused by: java.lang.IllegalArgumentException: Non class based component metadata rules must all be added before class based ones.
```

When a distribution and plugins are contributing component metadata rules, this prevents consumers and other plugins from using legacy component metadata rules.

**Comment by ljacomet** [comment]
Relevant Gradle issue: gradle/gradle#6568 and fix PR: gradle/gradle#6705

**Comment by ljacomet** [comment]
PR has been merged, next Gradle nightly, after 5.0-20180920235846+0000, will have the changes.

**Comment by rpalcolea** [comment]
We are able to use class-based component metadata rules. However, think we found an issue with cache. More details in -> https://github.com/nebula-plugins/gradle-nebula-integration/issues/42.

Closing this and moving conversation to new issue
